internal/backend/interface.go

```diff
@@ -45,9 +46,10 @@ type Application interface {
 	List(ctx context.Context, selector ApplicationSelector) ([]v1alpha1.Application, error)
 	Create(ctx context.Context, app *v1alpha1.Application) (*v1alpha1.Application, error)
 	Get(ctx context.Context, name string, namespace string) (*v1alpha1.Application, error)
-	Delete(ctx context.Context, name string, namespace string) error
+	Delete(ctx context.Context, name string, namespace string, deletionPropagationBackground bool) error
```

Comment:
How about instead of making this a bool to decide on foreground vs. background deletion, to make this into a dedicated type which follows current Kubernetes notations of background, foreground and orphan?

Reply:
@jannfis Done! If you wanted me to further rather the DeletionPropagation-style type in a DeletionOptions-style struct, let me know (my guess was that you did not).